src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -137,6 +140,16 @@ public void InstallWorkloadPack(PackInfo packInfo, SdkFeatureBand sdkFeatureBand
             }
         }
 
+        private PackageId GetPackageId(PackInfo packInfo)
+        {
+            var runtimeId = RuntimeInformation.RuntimeIdentifier;
+            var splitRuntimeId = RuntimeInformation.RuntimeIdentifier.Split('-');
+            var versionlessRuntimeId = $"{Regex.Replace(splitRuntimeId[0], @"[\d-]", string.Empty)}-{splitRuntimeId[1]}";
+            var packageId = packInfo.Aliases.ContainsKey(runtimeId) ? packInfo.Aliases[runtimeId].ToString() :
+                packInfo.Aliases.ContainsKey(versionlessRuntimeId) ? packInfo.Aliases[versionlessRuntimeId].ToString() : packInfo.Id;
```

Comment:
We shouldn't be using string splitting and regex matching to find the right alias here.  We should be doing the same thing that `WorkloadPack.TryGetAliasForRuntimeIdentifiers` is doing.  Ideally the PackInfo would just carry the resolved alias instead of all possible aliases, and we wouldn't have to re-run the logic, which is what I think @mhutch was suggesting.

Reply:
Ah, that makes more sense, I'll make that change.